File: solutions/erlang/hamming/1/src/hamming.erl
-module(hamming).

-export([distance/2]).


distance(Strand1, Strand2) -> distance(Strand1, Strand2, 0).

distance([Letter | Strand1], [Letter | Strand2], Distance) -> distance(Strand1, Strand2, Distance);
distance([_ | Strand1], [_ | Strand2], Distance) -> distance(Strand1, Strand2, Distance + 1);
distance([], [], Distance) -> Distance;
distance(_, _, _) -> {error, badarg}.

File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

// Distance calculates the Hamming Distance between two DNA strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("sequence lengths differ, therefore it is impossible to calculate hamming distance between them")
	}

	distance := 0

	// convert b in rune array
	// so it is index accessible
	bRune := []rune(b)

	for i, char := range a {
		if char != bRune[i] {
			distance++
		}
	}

	return distance, nil
}
